### DISCUSSION TITLE
Should we be checking the bookdown runs?

### DISCUSSION THREAD

**Discussion by alexnsaunders** [jtr13/cc21fall2#153] [Q&A]
Hi Team!

Should we be trying to resolve the bookdown issues - I saw one that was a white-space issue something like 
'name_of_markdown  .Rmd', which I corrected on main as such a quick and obvious fix

Right now I'm seeing from_dplyr_to_pandas_and_back is trying to call a local version of pandas - might be better using reticulate. For example - https://bookdown.org/yihui/rmarkdown-cookbook/eng-python.html
I'm unsure if we should be trying to correct - I pushed something else but the job is still failing, or if the assignee needs to request a change with the submitter or someone else is trouble-shooting these things..

Great work everyone on checking/assigning etc - I'm finding this a useful way to get to know github and hope you are too! :-)

**Comment by jtr13** [comment]
My two cents: in general we want to encourage the contributors to fix their own mistakes, a) since it gives them practice and b) since they were given specific instructions on what they were supposed to submit. That said, there are definitely situations in which it's easier to just fix it yourself and you'd prefer to do that--that's your judgment call. (It's only somewhat recently that maintainers are allowed to edit PRs directly, so that's kind of set up a tradition of requesting changes rather than implementing them oneself.)

There's a second category of problems such as the one you've described where we couldn't necessarily anticipate that the build would fail. It's up to you how much you want to invest in resolving the issue. It's always an option to ask the contributor to set `eval=FALSE` for all chunks and explain the reason at the beginning of the chapter. On this specific issue I recall that we weren't able to get Python running in a previous semester: What works locally doesn't necessarily work in the environment that GitHub Actions sets up to run R.

**Comment by jtr13** [comment]
Just realized I didn't answer the title question. Yes, you should check the builds and tag me if there's a problem. The builds were failing for a while and I'm in the process of straightening it out. I'll let you know when the builds should start passing. Please click the emoji button and react so I know who's getting these messages.
